src/CreatePackage.php

```diff
@@ -0,0 +1,181 @@
+
+        $name = $input->getArgument('name');
+        if (! preg_match('/^(zend|zf)-[a-z0-9-]+$/', $name)) {
+            throw new \InvalidArgumentException(
+                'Invalid package name, must be prefixed with "zend-" or "zf-"'
+                . ' and can contain only alphanumeric characters and dash.'
+            );
+        }
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $repo = $input->getArgument('name');
+
+        $errOutput = $output instanceof ConsoleOutputInterface ? $output->getErrorOutput() : $output;
+
+        if (is_dir($repo)) {
+            $errOutput->writeln(sprintf(
+                '<error>Directory with name "%s" already exists</error>',
+                $repo
+            ));
+
+            return;
+        }
+
+        if (! mkdir($repo, 0775) && ! is_dir($repo)) {
+            $errOutput->writeln(sprintf(
+                '<error>Cannot create package directory "%s"</error>',
+                $repo
+            ));
+
+            return;
+        }
+
+        $namespace = $this->determineNamespace($input, $output, $repo);
+
+        $replacement = [
+            '{org}' => $this->getOrg($repo),
+            '{repo}' => $repo,
+            '{category}' => $this->getCategory($repo),
+            '{namespace}' => $namespace,
+            '{namespace-test}' => preg_replace('/^(Zend|ZF)/', '\\1Test', $namespace),
+            '{year}' => date('Y'),
+        ];
+
+        $this->copyFiles(__DIR__ . '/../template/', $repo . '/', $replacement);
+
+        $helper = $this->getHelper('question');
+        $question = new ConfirmationQuestion('Would you like to create initial commit in the new package? [<info>Y/n</info>] ');
+
+        if ($helper->ask($input, $output, $question)) {
+            exec(sprintf(
+                'cd %s && git init . && git add . && git commit -am "Initial creation"',
+                $repo
+            ));
+        }
+    }
+
+    private function determineNamespace(InputInterface $input, OutputInterface $output, $repo)
+    {
+        $exp = explode('-', $repo);
+
+        $namespace = array_shift($exp) === 'zend' ? 'Zend' : 'ZF';
+
+        if ($exp[0] === 'expressive') {
+            $namespace .= '\Expressive';
+            array_shift($exp);
+        }
+
+        if (count($exp) === 1) {
+            $namespace .= '\\' . ucfirst(array_shift($exp));
+        }
+
+        if (! $exp) {
+            return $namespace;
+        }
+
+        $namespace1 = str_replace(' ', '\\', ucwords(implode(' ', $exp)));
+        $namespace2 = str_replace('\\', '', $namespace1);
+
+        $helper = $this->getHelper('question');
+
+        $question = new ChoiceQuestion(
+            'What namespace would you like to use?',
+            [
+                $namespace . '\\' . $namespace1,
+                $namespace . '\\' . $namespace2,
+            ]
+        );
+        $question->setErrorMessage('Invalid choice "%s". Please try again.');
```

Comment:
I think we should allow asking the user to _enter a custom namespace_. The reason is that some package names represent compound words which will translate to MixedCase namespace segments. As an example: zend-expressive-basicauth might translate to `Zend\Expressive\BasicAuth`, which would not be an option your code above will provide.